checker/src/abstract_value.rs

```diff
@@ -1148,7 +1156,8 @@ impl AbstractValueTrait for Rc<AbstractValue> {
             Expression::Widen { .. } => self.clone(),
 
             _ => self.clone(),
-        }
+        };
+        AbstractValue::make_with_tags(&result, self.tags.borrow().as_ref())
```

Comment:
This will become unnecessary with a dedicated kind of expression for tagged values.

Reply:
Added todo comments.